virtcontainers/mount.go

```diff
@@ -248,10 +248,16 @@ func bindMount(ctx context.Context, source, destination string, readonly bool) e
 
 	if err := ensureDestinationExists(absSource, destination); err != nil {
 		return fmt.Errorf("Could not create destination mount point %v: %v", destination, err)
-	} else if err := syscall.Mount(absSource, destination, "bind", syscall.MS_BIND, ""); err != nil {
+	}
+
+	if err := syscall.Mount(absSource, destination, "bind", syscall.MS_BIND, ""); err != nil {
 		return fmt.Errorf("Could not bind mount %v to %v: %v", absSource, destination, err)
 	}
 
+	if err := syscall.Mount("none", destination, "", syscall.MS_PRIVATE, ""); err != nil {
```

Comment:
Just curious to really understand how this solves the problem, and how is the PRIVATE flag helping?

Reply:
http://man7.org/linux/man-pages/man8/mount.8.html

Reply:
and http://man7.org/linux/man-pages/man2/mount.2.html